mlflow/models/docker_utils.py

```diff
@@ -13,9 +13,11 @@
 
 
 SETUP_PYENV_AND_VIRTUALENV = r"""# Setup pyenv
-RUN DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt-get -y install tzdata \
+RUN for i in 1 2 3; do \
+    DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt-get -y install tzdata \
     libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
-    libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+    libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev && break || sleep 30; \
```

Comment:
What is the actual cause of the intermittent failure? I'm not sure if we want users to do this looping for prod serving...

Reply:
For example here https://github.com/mlflow/mlflow/actions/runs/9342485139/job/25710677991#step:10:1345
I encountered such error a lot and normally I just rerun and it works. But I agree we shouldn't do this for prod serving, do you think we could add a param just for test to auto retry?

Reply:
Yea apparently it seems the error can happen due to network failure with various potential reasons, which might only happen in the CI. I vote for retrying the test itself for now, and we can investigate further once users report the same issue in their serving env.

Reply:
Yes it should be network error,

I propose to use retry with exponential backoffs, e.g., the first retry after 10s, the second retry after 20s, the third retry after 40s

Reply:
discussed offline with Serena. Updated `pyfunc_build_image` to retry a few times.